<!DOCTYPE html>
<html lang="en">
    <head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Galaxy Training!</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
    </head>
    <body>
        



<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i> Transcriptomics
                        </a>
                    </li>

                    
                        
                        
                        
                            <li class="nav-item dropdown">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Introduction slides">
                                    <i class="fa fa-slideshare" aria-hidden="true"></i> Introduction slides
                                </a>
                                <div class="dropdown-menu">
                                    
                                        
                                            
                                                <a class="dropdown-item" href="/training-material/topics/transcriptomics/slides/introduction.html">
                                                    Introduction to Transcriptomics
                                                </a>
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                </div>
                            </li>
                        
                    

                    

                    

                    
                    <li class="nav-item">
                        <a class="nav-link" href="https://zenodo.org/record/290221" title="Links to data">
                            <i class="fa fa-files-o" aria-hidden="true"></i> Input Dataset
                        </a>
                    </li>
                    

                    
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics#references" title="References">
                            <i class="fa fa-book" aria-hidden="true"></i> Literature
                        </a>
                    </li>
                    

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i> Help
    </a>
    <div class="dropdown-menu">
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Gitter
        </a>
        <a class="dropdown-item" href="https://biostar.usegalaxy.org/" title="Ask on Biostars">
            Biostars
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/ref-based/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <section class="tutorial">
        <h1>Reference-based RNA-Seq data analysis</h1>

        <blockquote class="overview">
            <h3>Overview</h3>

            <strong><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</strong>
            <ul>
            
            <li>What are the effects of Pasilla (PS) gene depletion on splicing events?</li>
            
            <li>How to analyze RNA sequencing data using a reference genome?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i> Objectives</strong>
            <ul>
            
            <li>Analysis of RNA sequencing data using a reference genome</li>
            
            <li>Analysis of differentially expressed genes</li>
            
            <li>Identification of functional enrichment among differentially expressed genes</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i> Requirements</strong>
            <ul>
            
                <li>
                    
                    <a href="/training-material/topics//introduction/">Galaxy introduction</a>
                    
                </li>
            
                <li>
                    
                    <a href="/training-material/topics//sequence-analysis/">Quality control</a>
                    
                </li>
            
            
            </ul>
            

            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i> Time estimation:</strong> 1d</p>
        </blockquote>

        <h1 class="no_toc" id="introduction">Introduction</h1>

<p>In the study of <a href="http://genome.cshlp.org/content/21/2/193.long">Brooks <em>et al.</em> 2011</a>, the Pasilla (PS) gene, <em>Drosophila</em> homologue of the Human splicing regulators Nova-1 and Nova-2 Proteins, was depleted in <em>Drosophila melanogaster</em> by RNAi. The authors wanted to identify exons that are regulated by Pasilla gene using RNA sequencing data.</p>

<p>Total RNA was isolated and used for preparing either single-end or paired-end RNA-seq libraries for treated (PS depleted) samples and untreated samples. These libraries were sequenced to obtain a collection of RNA sequencing reads for each sample. The effects of Pasilla gene depletion on splicing events can then be analyzed by comparison of RNA sequencing data of the treated (PS depleted) and the untreated samples.</p>

<p>The genome of <em>Drosophila melanogaster</em> is known and assembled. It can be used as reference genome to ease this analysis.  In a reference based RNA-seq data analysis, the reads are aligned (or mapped) against a reference genome, <em>Drosophila melanogaster</em> here, to significantly improve the ability to reconstruct transcripts and then identify differences of expression between several conditions.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li>
<a href="#pretreatments" id="markdown-toc-pretreatments">Pretreatments</a>    <ol>
      <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
      <li><a href="#quality-control" id="markdown-toc-quality-control">Quality control</a></li>
    </ol>
  </li>
  <li>
<a href="#mapping" id="markdown-toc-mapping">Mapping</a>    <ol>
      <li><a href="#preliminary-mapping" id="markdown-toc-preliminary-mapping">Preliminary mapping</a></li>
      <li><a href="#actual-mapping" id="markdown-toc-actual-mapping">Actual mapping</a></li>
      <li><a href="#inspection-of-hisat2-results" id="markdown-toc-inspection-of-hisat2-results">Inspection of HISAT2 results</a></li>
    </ol>
  </li>
  <li>
<a href="#analysis-of-the-differential-gene-expression" id="markdown-toc-analysis-of-the-differential-gene-expression">Analysis of the differential gene expression</a>    <ol>
      <li><a href="#count-the-number-of-reads-per-annotated-gene" id="markdown-toc-count-the-number-of-reads-per-annotated-gene">Count the number of reads per annotated gene</a></li>
      <li><a href="#analysis-of-the-differential-gene-expression-1" id="markdown-toc-analysis-of-the-differential-gene-expression-1">Analysis of the differential gene expression</a></li>
      <li><a href="#analysis-of-the-functional-enrichment-among-differentially-expressed-genes" id="markdown-toc-analysis-of-the-functional-enrichment-among-differentially-expressed-genes">Analysis of the functional enrichment among differentially expressed genes</a></li>
    </ol>
  </li>
  <li>
<a href="#inference-of-the-differential-exon-usage" id="markdown-toc-inference-of-the-differential-exon-usage">Inference of the differential exon usage</a>    <ol>
      <li><a href="#count-the-number-of-reads-per-exon" id="markdown-toc-count-the-number-of-reads-per-exon">Count the number of reads per exon</a></li>
      <li><a href="#differential-exon-usage" id="markdown-toc-differential-exon-usage">Differential exon usage</a></li>
    </ol>
  </li>
  <li><a href="#annotation-of-the-result-tables-with-gene-information" id="markdown-toc-annotation-of-the-result-tables-with-gene-information">Annotation of the result tables with gene information</a></li>
</ol>

</blockquote>

<h1 id="pretreatments">Pretreatments</h1>

<h2 id="data-upload">Data upload</h2>

<p>The original data is available at NCBI Gene Expression Omnibus (GEO) under accession number <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE18508">GSE18508</a>.</p>

<p>We will look at the 7 first samples:</p>

<ul>
  <li>3 treated samples with Pasilla (PS) gene depletion: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461179">GSM461179</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461180">GSM461180</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461181">GSM461181</a>
</li>
  <li>4 untreated samples: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461176">GSM461176</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461177">GSM461177</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461178">GSM461178</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461182">GSM461182</a>
</li>
</ul>

<p>Each sample constitutes a separate biological replicate of the corresponding condition (treated or untreated). Moreover, two of the treated and two of the untreated samples are from a paired-end sequencing assay, while the remaining samples are from a single-end sequencing experiment.</p>

<p>We have extracted sequences from the Sequence Read Archive (SRA) files to build FASTQ files.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-data-upload">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Data upload</h3>

  <ol>
    <li>Create a new history for this RNA-seq exercise</li>
    <li>Import a FASTQ file pair (<em>e.g.</em>  <code class="highlighter-rouge">GSM461177_untreat_paired_chr4_R1.fastq</code> and <code class="highlighter-rouge">GSM461177_untreat_paired_chr4_R2.fastq</code>)
      <ul>
        <li>Option 1: From a shared data library if available (ask your instructor)</li>
        <li>Option 2: From <a href="https://dx.doi.org/10.5281/zenodo.290221">Zenodo</a>
</li>
      </ul>

      <blockquote class="tip">
        <h3 id="-tip-importing-data-via-links">
<i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>Open the Galaxy Upload Manager</li>
          <li>Select <strong>Paste/Fetch Data</strong>
</li>
          <li>Paste the link into the text field</li>
          <li>Press <strong>Start</strong>
</li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Once the data file is in your history, verify that the datatype is <code class="highlighter-rouge">fastqsanger</code>, not <code class="highlighter-rouge">fastq</code>.
If the datatype is <code class="highlighter-rouge">fastq</code>, please change the file type to <code class="highlighter-rouge">fastqsanger</code></p>

      <blockquote class="tip">
        <h3 id="-tip-changing-the-datatype">
<i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Changing the datatype</h3>
        <ul>
          <li>Click on the pencil button displayed in your dataset in the history</li>
          <li>Choose <strong>Datatype</strong> on the top</li>
          <li>Select <code class="highlighter-rouge">fastqsanger</code>
</li>
          <li>Press <strong>Save</strong>
</li>
        </ul>
      </blockquote>
    </li>
    <li>Edit the “Database/Build” to select <code class="highlighter-rouge">dm3</code>
</li>
    <li>Rename the datasets according to the samples</li>
  </ol>

</blockquote>

<p>Both files contain the reads that belong to chromosome 4 of a paired-end sample. The sequences are raw sequences from the sequencing machine, without any pretreatments. They need to be controlled for their quality.</p>

<h2 id="quality-control">Quality control</h2>

<p>For quality control, we use similar tools as described in <a href="/training-material/topics/sequence-analysis">NGS-QC tutorial</a>: <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> and <a href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/">Trim Galore</a>.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-quality-control">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Quality control</h3>

  <ol>
    <li>
      <p><strong>FastQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run FastQC on both FASTQ files to control the quality of the reads</p>

      <blockquote class="question">
        <h3 id="-questions">
<i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <ol>
          <li>What is the read length?</li>
          <li>
            <p>Is there anything that you find striking when you compare the two reports?</p>

            <details>
<summary>Click to view answers</summary>
<ol type="1">
<li>The read length is 37 bp</li>
<li>Both reports for GSM461177_untreat_paired_chr4_R1 and for GSM461177_untreat_paired_chr4_R2 are quite ok. For GSM461177_untreat_paired_chr4_R1, there are several warnings and an issue with the Kmer content. For GSM461177_untreat_paired_chr4_R2, the quality in the 2nd tile is bad (maybe because of an issue during sequencing). We need to be careful for the quality treatment and to do it with paired-end information</li>
</ol>
</details>
          </li>
        </ol>
      </blockquote>
    </li>
    <li>
<strong>Trim Galore</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Treat for the quality of sequences by running Trim Galore
      <ul>
        <li>Indicate that we are dealing with <em>paired-end</em> datasets</li>
      </ul>

      <blockquote class="question">
        <h3 id="-questions-1">
<i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <p>Why is Trim Galore run once on the paired-end dataset and not twice on each dataset?</p>

        <details>
<summary>Click to view answers</summary>
Trim Galore can remove sequences if they become too short during the trimming process. For paired-end files Trim Galore! removes entire sequence pairs if one (or both) of the two reads became shorter than the set length cutoff. Reads of a read-pair that are longer than a given threshold but for which the partner read has become too short can optionally be written out to single-end files. This ensures that the information of a read pair is not lost entirely if only one read is of good quality.
</details>
      </blockquote>
    </li>
    <li>
      <p><strong>FastQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Re-run FastQC on Trim Galore’s outputs and inspect the differences</p>

      <blockquote class="question">
        <h3 id="-questions-2">
<i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <ol>
          <li>How did trimming affect the read lengths?</li>
          <li>
            <p>Are there other reported characteristics impacted by Trim Galore?</p>

            <details>
<summary>Click to view answers</summary>
<ol type="1">
<li>The read lengths are no longer uniform, but range now from 20 to 37 bp, *i.e.*, reads were trimmed to different extent.</li>
<li>For GSM461177_untreat_paired_chr4_R1, the per base sequence content is now red. For GSM461177_untreat_paired_chr4_R2, the per tile sequence quality is still bad but in addition now the per base sequence content and the Kmer Content are deemed problematic.</li>
</ol>
</details>
          </li>
        </ol>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>As the genome of <em>Drosophila melanogaster</em> is known and assembled, we can use this information and map the sequences on this genome to identify the effects of Pasilla gene depletion on splicing events.</p>

<h1 id="mapping">Mapping</h1>

<p>To make sense of the reads, their positions within the <em>Drosophila melanogaster</em> genome must be determined. This process is known as aligning or ‘mapping’ the reads to the reference genome.</p>

<blockquote>
  <h3 id="-comment">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

  <p class="comment">Do you want to learn more about the principles behind mapping? Follow our <a href="/training-material/topics/sequence-analysis/">training</a></p>
</blockquote>

<p>Because in the case of a eukaryotic transcriptome, most reads originate from processed mRNAs lacking introns, they cannot be simply mapped back to the genome as we normally do for DNA data. Instead the reads must be separated into two categories:</p>

<ul>
  <li>Reads that map entirely within exons</li>
  <li>Reads that cannot be mapped within an exon across their entire length because they span two or more exons</li>
</ul>

<p>Spliced mappers have been developed to efficiently map transcript-derived reads against genomes. <a href="https://ccb.jhu.edu/software/tophat/index.shtml">TopHat</a> was one of the first tools designed specifically to address these problems:</p>

<ol>
  <li>Identification of potential exons using reads that do map to the genome</li>
  <li>Generation of possible splices between neighboring exons</li>
  <li>Comparison of reads that did not initially map to the genome against these <em>in silico</em> created junctions</li>
</ol>

<figure id="figure-1"><img src="../../images/13059_2012_Article_3053_Fig6_HTML.jpg" alt="Kim et al." title="Kim et al., Genome Biology, 2013"><figcaption><span class="figcaption-prefix">Figure 1:</span> Kim et al., Genome Biology, 2013</figcaption></figure>

<p>Here, we will use HISAT2, a successor to TopHat2 that is faster with low memory requirements.</p>

<p>To conduct the mapping efficiently, HISAT2 needs to know one important parameter about the sequencing library: the library type.</p>

<p>This information should usually come with your FASTQ files, ask your sequencing facility! If not, try to find them on the site where you downloaded the data or in the corresponding publication. Another option is to estimate these parameters with a <em>preliminary mapping</em> of a <em>downsampled</em> file and some analysis programs. Afterward, the actual mapping can be redone on the original files with the optimized parameters.</p>

<h2 id="preliminary-mapping">Preliminary mapping</h2>

<p>In a preliminary mapping, we will estimate the library type to run HISAT2 efficiently afterwards.</p>

<blockquote class="comment">
  <h3 id="-comment-1">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>
  <p>This step is not necessary if you don’t need to estimate the library type of your data.</p>
</blockquote>

<p>The library type is determined by the library preparation protocol which, in turn, determines which of the two strands of cDNA obtained from the input RNA through reverse transcription ultimately get sequenced.</p>

<figure id="figure-2"><img src="../../images/strand_library_type.png" alt="Credit: Zhao Zhang" title="Credit: Zhao Zhang(http://onetipperday.sterding.com/2012/07/how-to-tell-which-library-type-to-use.html)"><figcaption><span class="figcaption-prefix">Figure 2:</span> Credit: Zhao Zhang(http://onetipperday.sterding.com/2012/07/how-to-tell-which-library-type-to-use.html)</figcaption></figure>

<p>In the previous illustration, you can see that in the dUTP method, for example, only the first cDNA strand, synthesized through reverse transcription of the original RNA, gets sequenced, while the second cDNA strand corresponding to the original RNA strand is degraded because of the dUTP incorporated into it.</p>

<table>
  <thead>
    <tr>
      <th>Examples of protocol</th>
      <th>Description</th>
      <th>Library Type (HISAT2)</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Standard Illumina</td>
      <td>Reads from the left-most end of the fragment (in transcript coordinates) map to the transcript strand, and the right-most end maps to the opposite strand</td>
      <td>Unstranded (default)</td>
    </tr>
    <tr>
      <td>dUTP, NSR, NNSR</td>
      <td>Same as above except we enforce the rule that the right-most end of the fragment (in transcript coordinates) is the first sequenced (or only sequenced for single-end reads). Equivalently, it is assumed that only the strand generated during first strand synthesis is sequenced.</td>
      <td>First strand (FR/F)</td>
    </tr>
    <tr>
      <td>Ligation, Standard SOLiD</td>
      <td>Same as above except we enforce the rule that the left-most end of the fragment (in transcript coordinates) is the first sequenced (or only sequenced for single-end reads). Equivalently, it is assumed that only the strand generated during second strand synthesis is sequenced.</td>
      <td>Second strand (RF/R)</td>
    </tr>
  </tbody>
</table>

<p>If you do not know the library type, you can find it by yourself by mapping the reads on the reference genome and infer the library type from the mapping results by comparing reads mapping information to the annotation of the reference genome.</p>

<figure id="figure-3"><img src="../../images/library_type_mapping.png" alt="Type of library, depending also of the type of sequencing" title="Type of library, depending also of the type of sequencing"><figcaption><span class="figcaption-prefix">Figure 3:</span> Type of library, depending also of the type of sequencing</figcaption></figure>

<p>Sequencing proceeds from 5’ to 3’. So, in the First Strand case, all reads from the left-most end of an RNA fragment (always from 5’ to 3’) are mapped to the transcript-strand, and (for pair-end sequencing) reads from the right-most end are always mapped to the opposite strand.</p>

<p>We can now try to determine the library type of our data.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-determining-the-library-type-optional">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Determining the library type (Optional)</h3>

  <ol>
    <li>Load the Ensembl gene annotation for <em>Drosophila melanogaster</em> (<a href="https://zenodo.org/record/290221/files/Drosophila_melanogaster.BDGP5.78.gtf"><code class="highlighter-rouge">Drosophila_melanogaster.BDGP5.78.gtf</code></a>) from the shared data library or from <a href="https://dx.doi.org/10.5281/zenodo.290221">Zenodo</a> into your current Galaxy history
      <ul>
        <li>Rename the dataset if necessary</li>
        <li>Verify that the datatype is <code class="highlighter-rouge">gtf</code> and not <code class="highlighter-rouge">gff</code>. If the datatype is not <code class="highlighter-rouge">gtf</code>, please change it using the pencil icon.</li>
      </ul>
    </li>
    <li>
      <p><strong>Select first</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Use the tool <strong>Select first - lines from a file</strong>  to downsample both FASTQ files generated by Trim Galore to 200k or 1M reads</p>

      <blockquote class="question">
        <h3 id="-questions-3">
<i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <ol>
          <li>Why are we downsampling our data here?</li>
          <li>How many rows must be selected to conserve 200k reads?</li>
        </ol>

        <details>
<summary>Click to view answers</summary>
<ol>
<li> We are performing a preliminary mapping for the purpose of determining the library type. We do not need the full dataset for this, and
     mapping the full dataset can take quite some time and resources, so it is better to perform this test on a small subset of the data.
</li>
<li> In a FASTQ file, a read corresponds to 4 lines. So to conserve 200,000 reads, 800,000 must be selected.
     More details about the FASTQ format can be found <a href="https://en.wikipedia.org/wiki/FASTQ_format">here</a>.
</li>
</ol>

</details>
      </blockquote>
    </li>
    <li>
<strong>HISAT2</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>HISAT2</strong> with:
      <ul>
        <li>“Source for the reference genome” to <code class="highlighter-rouge">Use a built-in genome</code>
</li>
        <li>“Reference genome” to <code class="highlighter-rouge">dm3</code>
</li>
        <li>“Single-end or parired-end reads?” to <code class="highlighter-rouge">paired-end</code>
</li>
        <li>“FASTA/Q file #1” to the downsampled <code class="highlighter-rouge">Trimmed reads pair 1</code> (Trim Galore output)</li>
        <li>“FASTA/Q file #2” to the downsampled <code class="highlighter-rouge">Trimmed reads pair 2</code> (Trim Galore output)</li>
        <li>Default values for other parameters</li>
      </ul>
    </li>
    <li>
<strong>Infer Experiment</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>Infer Experiment</strong> to determine the library type:
      <ul>
        <li>HISAT2 output as “Input BAM/SAM file”</li>
        <li>Drosophila annotation as “Reference gene model”</li>
      </ul>
    </li>
    <li>
      <p>Check the results and search the tool’s documentation for help on the meaning</p>

      <blockquote class="comment">
        <h3 id="-comment-2">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>
        <p>As it is sometimes quite difficult to find out which settings correspond to those of other programs, the following table might be helpful to identify the library type:</p>

        <table>
          <thead>
            <tr>
              <th>Sequencing type</th>
              <th><strong>Infer Experiment</strong></th>
              <th><strong>TopHat</strong></th>
              <th><strong>HISAT2</strong></th>
              <th><strong>htseq-count</strong></th>
              <th><strong>featureCounts</strong></th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>Paired-End (PE)</td>
              <td>“1++,1–,2+-,2-+”</td>
              <td>“FR Second Strand”</td>
              <td>“Second Strand F/FR”</td>
              <td>“yes”</td>
              <td>“1”</td>
            </tr>
            <tr>
              <td>PE</td>
              <td>“1+-,1-+,2++,2–”</td>
              <td>“FR First Strand”</td>
              <td>“First Strand R/RF”</td>
              <td>“reverse”</td>
              <td>“2”</td>
            </tr>
            <tr>
              <td>Single-End (SE)</td>
              <td>”++,–”</td>
              <td>“FR Second Strand”</td>
              <td>“Second Strand F/FR”</td>
              <td>“yes”</td>
              <td>“1”</td>
            </tr>
            <tr>
              <td>SE</td>
              <td>”+-,-+”</td>
              <td>“FR First Strand”</td>
              <td>“First Strand R/RF”</td>
              <td>“reverse”</td>
              <td>“2”</td>
            </tr>
            <tr>
              <td>SE,PE</td>
              <td>undecided</td>
              <td>“FR Unstranded”</td>
              <td>default</td>
              <td>“no”</td>
              <td>“0”</td>
            </tr>
          </tbody>
        </table>

      </blockquote>

      <blockquote class="question">
        <h3 id="-question">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

        <ol>
          <li>Which fraction of the reads in the BAM file can be explained assuming which library type?</li>
          <li>
            <p>Which library type do you choose? What is the corresponding term for this library type in <strong>HISAT2</strong>?</p>

            <details>
<summary>Click to view answer</summary>
<ol type="1">
<li>Fraction of reads explained by "1++,1--,2+-,2-+": 0.0151 and Fraction of reads explained by "1+-,1-+,2++,2--": 0.9843</li>
<li>The library seems to be of the type "1+-,1-+,2++,2--", which is called First Strand (R/RF) type in HISAT2, and "reverse" in htseq-count. </li>
</ol>
</details>
          </li>
        </ol>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="actual-mapping">Actual mapping</h2>

<p>We can now map all the RNA sequences on the <em>Drosophila melanogaster</em> genome using HISAT2.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-spliced-mapping">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Spliced mapping</h3>

  <ol>
    <li>
<strong>HISAT2</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>HISAT2</strong> with:
      <ul>
        <li>“Source for the reference genome” to <code class="highlighter-rouge">Use a built-in genome</code>
</li>
        <li>“Reference genome” to <code class="highlighter-rouge">dm3</code>
</li>
        <li>“Single-end or parired-end reads?” to <code class="highlighter-rouge">paired-end</code>
</li>
        <li>“FASTA/Q file #1” to <code class="highlighter-rouge">Trimmed reads pair 1</code> (Trim Galore output)</li>
        <li>“FASTA/Q file #2” to <code class="highlighter-rouge">Trimmed reads pair 2</code> (Trim Galore output)</li>
        <li>“Specify strand information” set to the previously determined value (<code class="highlighter-rouge">RF</code>)</li>
        <li>Default values for other parameters except “Spliced alignment options”</li>
        <li>“Disable spliced alignment” to <code class="highlighter-rouge">False</code>
</li>
        <li>“GTF file with known splice sites” to <code class="highlighter-rouge">Drosophila_melanogaster.BDGP5.78.gtf</code>
</li>
      </ul>
    </li>
    <li>Inspect the mapping statistics
      <ul>
        <li>Click on “View details” (“i” icon)</li>
        <li>Click on “stderr” (Tool Standard Error)</li>
      </ul>

      <blockquote class="question">
        <h3 id="-question-1">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

        <ol>
          <li>How many paired reads were mapped 1 time? And how many paired reads were mapped more than 1 time?</li>
          <li>How many reads were mapped but without their mate?</li>
          <li>
            <p>What is the overall alignment rate?</p>

            <details>
<summary>Click to view answer</summary>
<ol type="1">
<li>37.84% and 14.69%</li>
<li>14,454 reads (the reads aligned discordantly 1 time)</li>
<li>The overall alignment rate is 93.52%. It counts proportion of mapped reads: (15413+5985+14454+3637/2+849/2)/40736</li>
</ol>
</details>
          </li>
        </ol>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p><strong>HISAT2</strong> generates a BAM file with the mapped reads.</p>

<blockquote class="question">
  <h3 id="-question-2">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

  <ol>
    <li>What is a BAM file?</li>
    <li>
      <p>What does such a file contain?</p>

      <details>
<summary>Click to view answer</summary>
<ol type="1">
<li>a BAM file is the binary version of a SAM file</li>
<li>It contains information about the mapping: for each mapped read, the position on the reference genome, the mapping quality, ...</li>
</ol>
</details>
    </li>
  </ol>
</blockquote>

<p>The mapping exercise worked for you? Great! <img class="emoji" title=":tada:" alt=":tada:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f389.png" height="20" width="20"></p>

<blockquote class="hands_on">
  <h3 id="-optional-hands-on-map-other-datasets">
<i class="fa fa-pencil" aria-hidden="true"></i> (Optional) Hands-on: Map other datasets</h3>

  <p>You can do the same process on the other sequence files available on <a href="https://dx.doi.org/10.5281/zenodo.290221">Zenodo</a></p>

  <ul>
    <li>Paired-end data
      <ul>
        <li>
<code class="highlighter-rouge">GSM461178_untreat_paired_chr4_R1</code> and <code class="highlighter-rouge">GSM461178_untreat_paired_chr4_R2</code>
</li>
        <li>
<code class="highlighter-rouge">GSM461180_treat_paired_chr4_R1</code> and <code class="highlighter-rouge">GSM461180_treat_paired_chr4_R2</code>
</li>
        <li>
<code class="highlighter-rouge">GSM461181_treat_paired_chr4_R1</code> and <code class="highlighter-rouge">GSM461181_treat_paired_chr4_R2</code>
</li>
      </ul>
    </li>
    <li>Single-end data
      <ul>
        <li><code class="highlighter-rouge">GSM461176_untreat_single_chr4</code></li>
        <li><code class="highlighter-rouge">GSM461179_treat_single_chr4</code></li>
        <li><code class="highlighter-rouge">GSM461182_untreat_single_chr4</code></li>
      </ul>
    </li>
  </ul>

  <p>This is really interesting to redo on the other datasets, specially to check how the parameters are inferred given the different type of data.</p>
</blockquote>

<h2 id="inspection-of-hisat2-results">Inspection of HISAT2 results</h2>

<p>The BAM file contains information about where the reads are mapped on the reference genome. But it is a binary file and with the information for more than 3 million reads encoded in it, it is difficult to inspect and explore the file.</p>

<p>A powerful tool to visualize the content of BAM files is the Integrative Genomics Viewer IGV.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-inspection-of-hisat2-results">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Inspection of HISAT2 results</h3>

  <ol>
    <li>
<strong>IGV</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Visualize the aligned reads
      <ul>
        <li>Click on the HISAT2 BAM output in your history to expand it.</li>
        <li>Towards the bottom of the history item, find the line starting with <code class="highlighter-rouge">Display with IGV</code>. This is followed by 2 links:
          <ul>
            <li>option 1: <code class="highlighter-rouge">local</code>. Select this option if you already have IGV installed on your machine.</li>
            <li>option 2: <code class="highlighter-rouge">D. melanogaster (dm3)</code>. This will download and launch IGV on your local machine.</li>
          </ul>
        </li>
        <li>Once IGV has started, navigate to chromosome 4 between 560 kb to 600 kb (<code class="highlighter-rouge">chr4:560,000-600,000</code>)</li>
      </ul>

      <blockquote class="comment">
        <h3 id="-comments">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>

        <ul>
          <li>In order for this step to work, you will need to have either IGV or <a href="https://www.java.com/en/download/faq/java_webstart.xml">Java web start</a>
installed on your machine. However, the questions in this section can also be answered by inspecting the IGV screenshots below.</li>
          <li>Check the <a href="https://software.broadinstitute.org/software/igv/AlignmentData">IGV documentation</a> for more information.</li>
        </ul>

      </blockquote>

      <blockquote class="question">
        <h3 id="-question-3">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

        <p>In the following screenshot,</p>
        <ol>
          <li>Which information does appear on the top in grey?</li>
          <li>
            <p>What do the connecting lines between some of the aligned reads indicate?</p>

            <figure id="figure-4"><img src="../../images/junction_igv_screenshot.png" alt="Screenshot of the IGV view on Chromosome 4" title="Screenshot of IGV on Chromosome 4"><figcaption><span class="figcaption-prefix">Figure 4:</span> Screenshot of IGV on Chromosome 4</figcaption></figure>

            <details>
<summary>Click to view answers</summary>
<ol type="1">
<li>The coverage plot: the sum of mapped reads at each position</li>
<li>They indicate junction events (or splice sites), *i.e.*, reads that are mapped across an intron</li>
</ol>
</details>
          </li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p><strong>IGV</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Zoom to <code class="highlighter-rouge">chr4:560,000-600,000</code> and inspect the splice junctions using a <strong>Sashimi plot</strong></p>

      <blockquote class="tip">
        <h3 id="-tip-creation-of-a-sashimi-plot">
<i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Creation of a Sashimi plot</h3>

        <ul>
          <li>Right click on the BAM file</li>
          <li>Select <strong>Sashimi Plot</strong> from the context menu</li>
        </ul>
      </blockquote>

      <blockquote class="question">
        <h3 id="-question-4">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

        <figure id="figure-5"><img src="../../images/hisat_igv_sashimi.png" alt="Screenshot of a Sashimi plot of Chromosome 4" title="Screenshot of a Sashimi plot of Chromosome 4"><figcaption><span class="figcaption-prefix">Figure 5:</span> Screenshot of a Sashimi plot of Chromosome 4</figcaption></figure>

        <ol>
          <li>What does the vertical bar graph represent? And the numbered arcs?</li>
          <li>What do the numbers on the arcs mean?</li>
          <li>
            <p>Why do we observe 4 different stacked groups of blue linked boxes at the bottom right?</p>

            <details>
<summary>Click to view answers</summary>
<ol type="1">
<li>The coverage for each alignment track is plotted as a bar graph. Arcs represent observed splice junctions, *i.e.*, reads spanning introns</li>
<li>The numbers refer to the number of these observed junction reads. </li>
<li>The 4 groups of linked boxes on the bottom represent 4 transcripts from a single gene differing in the first exon.</li>
</ol>
</details>
          </li>
        </ol>
      </blockquote>

      <blockquote class="comment">
        <h3 id="-comment-3">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

        <p>Check the <a href="https://software.broadinstitute.org/software/igv/Sashimi">IGV documentation on Sashimi plots</a> to find some clues</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>After the mapping, we have in the generated mapping file the information about where the reads are mapped on the reference genome. So for each mapped read, we know where it is mapped and how good it was mapped.</p>

<p>The next step in the RNA-Seq data analysis is quantification of expression level of the genomic features (gene, transcript, exons, …) to be able then to compare several samples for the different expression analysis. The quantification consist into taking each known genomic feature (<em>e.g.</em> gene) of the reference genome and then counting how many reads are mapped on this genomic feature. So, in this step, we start with an information per mapped reads to end with an information per genomic feature.</p>

<blockquote class="comment">
  <h3 id="-comment-4">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

  <p>The quantification depends on the definition of the genomic features of the reference genome, and then on the annotations. We strongly recommend you to use an annotation corresponding to the same version of the reference genome you used for the mapping.</p>
</blockquote>

<p>To identify exons that are regulated by the Pasilla gene, we need to identify genes and exons which are differentially expressed between samples with PS gene depletion and control samples.
In this tutorial, we will then analyze the differential gene expression, but also the differential exon usage.</p>

<h1 id="analysis-of-the-differential-gene-expression">Analysis of the differential gene expression</h1>

<p>We will first investigate the differential gene expression to identify which genes are impacted by the Pasilla gene depletion</p>

<h2 id="count-the-number-of-reads-per-annotated-gene">Count the number of reads per annotated gene</h2>

<p>To compare the expression of single genes between different conditions (<em>e.g.</em> with or without PS depletion), an essential first step is to quantify the number of reads per gene. <a href="https://www-huber.embl.de/users/anders/HTSeq/doc/count.html"><strong>HTSeq-count</strong></a> is one of the most popular tools for gene quantification.</p>

<p>To quantify the number of reads mapped to a gene, an annotation of the gene position is needed. In a previous step, we have already uploaded the <code class="highlighter-rouge">Drosophila_melanogaster.BDGP5.78.gtf</code> with the Ensembl gene annotation for <em>Drosophila melanogaster</em> to Galaxy, which we can now make use of for this purpose.</p>

<p>In principle, the counting of reads overlapping with genomic features is a fairly simple task. But there are some details that need to be decided, such how to handle multi-mapping reads. <strong>HTSeq-count</strong> offers 3 choices (“union”, “intersection_strict” and “intersection_nonempty”) to handle read mapping to multiple locations, reads overlapping introns, or reads that overlap more than one genomic feature:</p>

<figure id="figure-6"><img src="../../images/htseq_count.png" alt="HT-Seq methods to handle overlapping reads" title="HT-Seq methods to handle overlapping reads (HTSeq documentation: https://www-huber.embl.de/users/anders/HTSeq/doc/count.html)"><figcaption><span class="figcaption-prefix">Figure 6:</span> HT-Seq methods to handle overlapping reads (HTSeq documentation: https://www-huber.embl.de/users/anders/HTSeq/doc/count.html)</figcaption></figure>

<p>The recommended mode is “union”, which counts overlaps even if a read only shares parts of its sequence with a genomic feature and disregards reads that overlap more than one feature.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-counting-the-number-of-reads-per-annotated-gene">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Counting the number of reads per annotated gene</h3>

  <ol>
    <li>
<strong>HTSeq-count</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>HTSeq-count</strong> on the BAM file with
      <ul>
        <li>
<code class="highlighter-rouge">Drosophila_melanogaster.BDGP5.78.gtf</code> as “GFF file”</li>
        <li>The “Union” mode</li>
        <li>A “Minimum alignment quality” of 10</li>
        <li>“Stranded” to <code class="highlighter-rouge">reverse</code>
</li>
      </ul>
    </li>
    <li>
      <p>Inspect the result files</p>

      <blockquote class="question">
        <h3 id="-question-5">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

        <ol>
          <li>Which information does the result file contain?</li>
          <li>
            <p>Which feature has the most reads mapped on it?</p>

            <details>
<summary>Click to view answers</summary>
<ol type="1">
<li>The useful result file is a tabular file with two columns: the gene id and the number of reads mapped on the corresponding gene</li>
<li>To display the most abundantly detected feature, we need to sort the output file with the features and the number of reads mapped to them. This can be done using the Sort tool on the second column and in descending order, which reveals that FBgn0017545 is the feature with the most reads (7,650) mapped on it.</li>
</ol>
</details>
          </li>
        </ol>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="analysis-of-the-differential-gene-expression-1">Analysis of the differential gene expression</h2>

<p>In the previous section, we counted only reads that mapped to genes of chromosome 4 and for only one sample. To be able to identify differential gene expression induced by PS depletion, all datasets (3 treated and 4 untreated) must be analyzed following the same procedure and for the whole genome.</p>

<p>To save time, we have run the necessary steps for you and obtained 7 count files, available on <a href="https://dx.doi.org/10.5281/zenodo.290221">Zenodo</a>.</p>

<p>These files contain for each gene of Drosophila the number of reads mapped to it. We could compare the files directly and calculate the extent of differential gene expression, but the number of sequenced reads mapped to a gene depends on:</p>

<ul>
  <li>Its own expression level</li>
  <li>Its length</li>
  <li>The sequencing depth of the sample</li>
  <li>The expression of all other genes within the sample</li>
</ul>

<p>Either for within- or for between-sample comparison, the gene counts need to be normalized. We can then use the Differential Gene Expression (DGE) analysis, whose two basic tasks are:</p>

<ul>
  <li>Estimate the biological variance using the replicates for each condition</li>
  <li>Estimate the significance of expression differences between any two conditions</li>
</ul>

<p>This expression analysis is estimated from read counts and attempts are made to correct for variability in measurements using replicates that are absolutely essential for accurate results. For your own analysis, we advice you to use at least 3, but preferably 5 biological replicates per condition. You can have different number of replicates per condition.</p>

<p><a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html"><strong>DESeq2</strong></a> is a great tool for DGE analysis. It takes read counts produced by <strong>HTseq-count</strong>, combines them into a big table (with genes in the rows and samples in the columns) and applies size factor normalization:</p>

<ul>
  <li>Computation for each gene of the geometric mean of read counts across all samples</li>
  <li>Division of every gene count by the geometric mean</li>
  <li>Use of the median of these ratios as a sample’s size factor for normalization</li>
</ul>

<p>Multiple factors with several levels can then be incorporated in the analysis. After normalization we can compare, in a statistically reliable way, the response of the expression of any gene to the presence of different levels of a factor.</p>

<p>In our example, we have samples with two varying factors that can explain differences in gene expression:</p>

<ul>
  <li>Treatment (either treated or untreated)</li>
  <li>Sequencing type (paired-end or single-end)</li>
</ul>

<p>Here treatment is the primary factor which we are interested in. The sequencing type is some further information that we know about the data that might affect the analysis. This particular multi-factor analysis allows us to assess the effect of the treatment, while taking the sequencing type into account, too.</p>

<blockquote class="comment">
  <h3 id="-comment-5">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

  <p>We recommend you to add as many factors as you think may affect gene expression in your experiment. It can be the sequencing type like here, but it can also be the manipulation (if different persons are involved in the library preparation), …</p>
</blockquote>

<blockquote class="hands_on">
  <h3 id="-hands-on-analysis-of-the-differential-gene-expression-1">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Analysis of the differential gene expression (1)</h3>

  <ol>
    <li>Create a new history</li>
    <li>Import the seven count files from <a href="https://dx.doi.org/10.5281/zenodo.290221">Zenodo</a>
      <ul>
        <li><code class="highlighter-rouge">GSM461176_untreat_single.deseq.counts</code></li>
        <li><code class="highlighter-rouge">GSM461177_untreat_paired.deseq.counts</code></li>
        <li><code class="highlighter-rouge">GSM461178_untreat_paired.deseq.counts</code></li>
        <li><code class="highlighter-rouge">GSM461179_treat_single.deseq.counts</code></li>
        <li><code class="highlighter-rouge">GSM461180_treat_paired.deseq.counts</code></li>
        <li><code class="highlighter-rouge">GSM461181_treat_paired.deseq.counts</code></li>
        <li><code class="highlighter-rouge">GSM461182_untreat_single.deseq.counts</code></li>
      </ul>
    </li>
    <li>
<strong>DESeq2</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>DESeq2</strong> with:
      <ul>
        <li>
          <p>“Treatment” as first factor with “treated” and “untreated” as levels and selection of count files corresponding to both levels</p>

          <blockquote class="tip">
            <h3 id="-tip">
<i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip</h3>

            <p>You can select several files by keeping the CTRL (or COMMAND) key pressed and clicking on the interesting files</p>
          </blockquote>
        </li>
        <li>
          <p>“Sequencing” as second factor with “PE” and “SE” as levels and selection of count files corresponding to both levels</p>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="-comment-6">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

        <p>File names have all information needed</p>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p>The first output of <strong>DESeq2</strong> is a tabular file. The columns are:</p>

<ol>
  <li>Gene identifiers</li>
  <li>Mean normalized counts, averaged over all samples from both conditions</li>
  <li>
    <p>Logarithm (to basis 2) of the fold change</p>

    <p>The log2 fold changes are based on primary factor level 1 vs. factor level 2, hence the order of factor levels is important. For example, for the factor ‘Treatment’, DESeq2 computes fold changes of ‘treated’ samples against ‘untreated’, <em>i.e.</em> the values correspond to up- or downregulation of genes in treated samples.</p>
  </li>
  <li>Standard error estimate for the log2 fold change estimate</li>
  <li>
<a href="https://en.wikipedia.org/wiki/Wald_test">Wald</a> statistic</li>
  <li>
<em>p</em>-value for the statistical significance of this change</li>
  <li>
<em>p</em>-value adjusted for multiple testing with the Benjamini-Hochberg procedure which controls false discovery rate (<a href="https://en.wikipedia.org/wiki/False_discovery_rate">FDR</a>)</li>
</ol>

<blockquote class="hands_on">
  <h3 id="-hands-on-analysis-of-the-differential-gene-expression-2">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Analysis of the differential gene expression (2)</h3>

  <ol>
    <li>
      <p><strong>Filter</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>Filter</strong> to extract genes with a significant change in gene expression (adjusted <em>p</em>-value below 0.05) between treated and untreated samples</p>

      <blockquote class="question">
        <h3 id="-question-6">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

        <p>How many genes have a significant change in gene expression between these conditions?</p>

        <details>
<summary>Click to view answers</summary>
To filter, you need to add the expression "c7&lt;0.05". And we get 751 genes (5.05%) with a significant change in gene expression between treated and untreated samples.
</details>
      </blockquote>

      <blockquote class="comment">
        <h3 id="-comment-7">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

        <p>The file with the independent filtered results can be used for further downstream analysis as it excludes genes with only few read counts as these genes will not be considered as significantly differentially expressed.</p>
      </blockquote>
    </li>
    <li>
      <p><strong>Filter</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Extract genes that are significantly up and downregulated in treated samples</p>

      <blockquote class="comment">
        <h3 id="-comments-1">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>
        <p>Rename your datasets for the downstream analyses</p>
      </blockquote>

      <blockquote class="question">
        <h3 id="-question-7">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

        <p>Are there more upregulated or downregulated genes in the treated samples?</p>

        <details>
<summary>Click to view answers</summary>
To obtain the up-regulated genes, we filter the previously generated file (with the significant change in gene expression) with the expression "c3&gt;0" (the log2 fold changes must be greater than 0). We obtain 331 genes (44.07% of the genes with a significant change in gene expression). For the down-regulated genes, we did the inverse and we 420 genes (55.93% of the genes with a significant change in gene expression)
</details>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p>In addition to the list of genes, <strong>DESeq2</strong> outputs a graphical summary of the results, useful to evaluate the quality of the experiment:</p>

<ol>
  <li>Histogram of <em>p</em>-values for all tests</li>
  <li>
<a href="https://en.wikipedia.org/wiki/MA_plot">MA plot</a>: global view of the relationship between the expression change of conditions (log ratios, M), the average expression strength of the genes (average mean, A), and the ability of the algorithm to detect differential gene expression. The genes that passed the significance threshold (adjusted p-value &lt; 0.1) are colored in red.</li>
  <li>
    <p>Principal Component Analysis (<a href="https://en.wikipedia.org/wiki/Principal_component_analysis">PCA</a>) and the first two axes</p>

    <p>Each replicate is plotted as an individual data point. This type of plot is useful for visualizing the overall effect of experimental covariates and batch effects.</p>

    <blockquote class="question">
      <h3 id="-questions-4">
<i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

      <ol>
        <li>What is the first axis separating?</li>
        <li>
          <p>And the second axis?</p>

          <details>
<summary>Click to view answers</summary>
<ol type="1">
<li>The first axis is seperating the treated samples from the untreated samples, as defined when DESeq2 was launched</li>
<li>The second axis is separating the single-end datasets from the paired-end datasets</li>
</ol>
</details>
        </li>
      </ol>
    </blockquote>
  </li>
  <li>
    <p>Heatmap of sample-to-sample distance matrix: overview over similarities and dissimilarities between samples</p>

    <blockquote class="question">
      <h3 id="-questions-5">
<i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

      <p>How are the samples grouped?</p>

      <details>
   <summary>Click to view answers</summary>
   <ol type="1">    
   <li>They are first grouped depending on the treatment (the first factor) and after on the library type (the second factor), as defined when DESeq2 was launched</li>
   </ol>
   </details>
    </blockquote>
  </li>
  <li>
    <p>Dispersion estimates: gene-wise estimates (black), the fitted values (red), and the final maximum a posteriori estimates used in testing (blue)</p>

    <p>This dispersion plot is typical, with the final estimates shrunk from the gene-wise estimates towards the fitted estimates. Some gene-wise estimates are flagged as outliers and not shrunk towards the fitted value. The amount of shrinkage can be more or less than seen here, depending on the sample size, the number of coefficients, the row mean and the variability of the gene-wise estimates.</p>
  </li>
</ol>

<p>For more information about <strong>DESeq2</strong> and its outputs, you can have a look at <a href="https://www.bioconductor.org/packages/release/bioc/manuals/DESeq2/man/DESeq2.pdf"><strong>DESeq2</strong> documentation</a>.</p>

<h2 id="analysis-of-the-functional-enrichment-among-differentially-expressed-genes">Analysis of the functional enrichment among differentially expressed genes</h2>

<p>We have extracted genes that are differentially expressed in treated (with PS gene depletion) samples compared to untreated samples. We would like to know the functional enrichment among the differentially expressed genes.</p>

<p>The Database for Annotation, Visualization and Integrated Discovery (<a href="https://david.ncifcrf.gov/">DAVID</a>) provides a comprehensive set of functional annotation tools for investigators to understand the biological meaning behind large lists of genes.</p>

<p>The query to DAVID can be done only on 100 genes. So, we will need to select the ones where the most interested in.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on:</h3>

  <ol>
    <li>
<strong>Sort</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Sort the 2 datasets generated previously (upregulated genes and downregulated genes) given the log2 fold change, in descending or ascending order (to obtain the higher absolute log2 fold changes on the top)</li>
    <li>
<strong>Select first lines from a dataset</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Extract the first 100 lines of sorted files</li>
    <li>
<strong>DAVID</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>DAVID</strong> on these files with
      <ul>
        <li>First column as “Column with identifiers”</li>
        <li>“ENSEMBL_GENE_ID” as “Identifier type”</li>
      </ul>

      <p>The output of the <strong>DAVID</strong> tool is a HTML file with a link to the DAVID website.</p>
    </li>
    <li>
      <p>Inspect the Functional Annotation Chart</p>

      <blockquote class="question">
        <h3 id="-questions-6">
<i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <p>What functional categories are the most represented?</p>

        <details>
<summary>Click to view answers</summary>
The up-regulated genes are mostly related to membrane (in the number of genes). The most represented functional categories are linked to signal and pathways for the down-regulated genes.
</details>
      </blockquote>
    </li>
    <li>
      <p>Inspect the Functional Annotation Clusterings</p>

      <blockquote class="question">
        <h3 id="-questions-7">
<i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <p>What functional annotations are the first clusters related to?</p>

        <details>
<summary>Click to view answers</summary>
For the up-regulated genes, the first cluster is more composed of functions related to chaperone and stress response. The down-regulated genes are more linked to ligase activity.
</details>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 id="inference-of-the-differential-exon-usage">Inference of the differential exon usage</h1>

<p>Next, we would like to know the differential exon usage between treated (PS depleted) and untreated samples using RNA-seq exon counts. We will rework on the mapping results we generated previously.</p>

<p>We will use <a href="https://www.bioconductor.org/packages/release/bioc/html/DEXSeq.html">DEXSeq</a>. DEXSeq detects high sensitivity genes, and in many cases exons, that are subject to differential exon usage. But first, as for the differential gene expression, we need to count the number of reads mapping to the exons.</p>

<h2 id="count-the-number-of-reads-per-exon">Count the number of reads per exon</h2>

<p>This step is similar to the step of <a href="#count-the-number-of-reads-per-annotated-gene">counting the number of reads per annotated gene</a> except that, instead of HTSeq-count, we are using DEXSeq-Count.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-counting-the-number-of-reads-per-exon">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Counting the number of reads per exon</h3>

  <ol>
    <li>
<strong>DEXSeq-Count</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Use the <strong>DEXSeq-Count</strong> to prepare the <em>Drosophila</em> annotations (<code class="highlighter-rouge">Drosophila_melanogaster.BDGP5.78.gtf</code>) to extract only exons with corresponding gene ids
      <ul>
        <li>“Prepare annotation” of “Mode of operation”</li>
      </ul>

      <p>The output is again a GTF file that is ready to be used for counting</p>
    </li>
    <li>
<strong>DEXSeq-Count</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Count reads using <strong>DEXSeq-Count</strong> with
      <ul>
        <li>HISAT2 output as “Input bam file”</li>
        <li>The formatted GTF file</li>
      </ul>
    </li>
    <li>
      <p>Inspect the result files</p>

      <blockquote class="question">
        <h3 id="-question-8">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

        <p>Which exon has the most reads mapped to it? From which gene has this exon been extracted? Is there a connection to the previous result obtained with HTSeq-count?</p>

        <details>
<summary>Click to view answers</summary>
FBgn0017545:004 is the exon with the most reads mapped to it. It is part of FBgn0017545, the feature with the most reads mapped with HTSeq-count
</details>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="differential-exon-usage">Differential exon usage</h2>

<p>DEXSeq usage is similar to DESeq2. It uses similar statistics to find differentially used exons.</p>

<p>As for DESeq2, in the previous step, we counted only reads that mapped to exons on chromosome 4 and for only one sample. To be able to identify differential exon usage induced by PS depletion, all datasets (3 treated and 4 untreated) must be analyzed following the same procedure. To save time, we did that for you. The results are available on <a href="https://dx.doi.org/10.5281/zenodo.290221">Zenodo</a>:</p>

<ul>
  <li>
<a href="https://zenodo.org/record/290221/files/dexseq.gtf">dexseq.gtf</a>: the results of running DEXSeq-count in ‘Prepare annotation’ mode</li>
  <li>Seven count files generated in ‘Count reads’ mode</li>
</ul>

<blockquote class="hands_on">
  <h3 id="-hands-on-1">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on:</h3>

  <ol>
    <li>Create a new history</li>
    <li>Import the seven count files and the dexseq.gtf from <a href="https://dx.doi.org/10.5281/zenodo.290221">Zenodo</a>
      <ul>
        <li><code class="highlighter-rouge">dexseq.gtf</code></li>
        <li><code class="highlighter-rouge">GSM461176_untreat_single.dexseq.counts</code></li>
        <li><code class="highlighter-rouge">GSM461177_untreat_paired.dexseq.counts</code></li>
        <li><code class="highlighter-rouge">GSM461178_untreat_paired.dexseq.counts</code></li>
        <li><code class="highlighter-rouge">GSM461179_treat_single.dexseq.counts</code></li>
        <li><code class="highlighter-rouge">GSM461180_treat_paired.dexseq.counts</code></li>
        <li><code class="highlighter-rouge">GSM461181_treat_paired.dexseq.counts</code></li>
        <li><code class="highlighter-rouge">GSM461182_untreat_single.dexseq.counts</code></li>
      </ul>
    </li>
    <li>
<strong>DEXSeq</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>DEXSeq</strong> with
      <ul>
        <li>“condition” as first factor with “treated” and “untreated” as levels and selection of count files corresponding to both levels</li>
        <li>“Sequencing” as second factor with “PE” and “SE” as levels and selection of count files corresponding to both levels</li>
      </ul>

      <blockquote class="comment">
        <h3 id="-comment-8">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

        <p>Unlike DESeq2, DEXSeq does not allow flexible primary factor names. Always use your primary factor name as “condition”</p>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p>Similarly to DESeq2, DEXSeq generates a table with:</p>

<ol>
  <li>Exon identifiers</li>
  <li>Gene identifiers</li>
  <li>Exon identifiers in the Gene</li>
  <li>Mean normalized counts, averaged over all samples from both conditions</li>
  <li>
    <p>Logarithm (to basis 2) of the fold change</p>

    <p>The log2 fold changes are based on primary factor level 1 vs. factor level 2. The order of factor levels is then important. For example, for the factor ‘Condition’, DESeq2 computes fold changes of ‘treated’ samples against ‘untreated’, <em>i.e.</em> the values correspond to up- or downregulations of genes in treated samples.</p>
  </li>
  <li>Standard error estimate for the log2 fold change estimate</li>
  <li>
<em>p</em>-value for the statistical significance of this change</li>
  <li>
<em>p</em>-value adjusted for multiple testing with the Benjamini-Hochberg procedure which controls false discovery rate (<a href="https://en.wikipedia.org/wiki/False_discovery_rate">FDR</a>)</li>
</ol>

<blockquote class="hands_on">
  <h3 id="-hands-on-2">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on:</h3>

  <ol>
    <li>
      <p><strong>Filter</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>Filter</strong> to extract exons with a significant differential usage (adjusted <em>p</em>-value equal or below 0.05) between treated and untreated samples</p>

      <blockquote class="question">
        <h3 id="-question-9">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

        <p>How many exons show a significant change in usage between these conditions?</p>

        <details>
<summary>Click to view answers</summary>
We get 38 exons (12.38%) with a significant usage change between treated and untreated samples
</details>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 id="annotation-of-the-result-tables-with-gene-information">Annotation of the result tables with gene information</h1>

<p>Unfortunately, in the process of counting, we loose all the information of the gene except its identifiant. In order to get the information back to our final counting tables, we can use a tool to make the correspondance between identifiant and annotation.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-3">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on:</h3>

  <ol>
    <li>
<strong>Annotate DE(X)Seq result</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>Annotate DE(X)Seq result</strong> on a counting table (from DESeq or DEXSeq) using the <code class="highlighter-rouge">Drosophila_melanogaster.BDGP5.78.gtf</code> as annotation file</li>
  </ol>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In this tutorial, we have analyzed real RNA sequencing data to extract useful information, such as which genes are up- or downregulated by depletion of the Pasilla gene and which genes are regulated by the Pasilla gene. To answer these questions, we analyzed RNA sequence datasets using a reference-based RNA-seq data analysis approach. This approach can be summarized with the following scheme:</p>

<figure id="figure-7"><img src="../../images/rna_quantification.png" alt="Summary of the analysis pipeline used" title="Summary of the analysis pipeline used"><figcaption><span class="figcaption-prefix">Figure 7:</span> Summary of the analysis pipeline used</figcaption></figure>


        
        <blockquote class="key_points">
            <h3>
<i class="fa fa-key" aria-hidden="true"></i> Key points</h3>

            <ul>
                
                <li>Using a spliced mapping tool for eukaryotic RNA seq data</li>
                
                <li>Running a differential gene expression with taking care of the factors to study</li>
                
                <li>Running a differential exon usage with taking care of the factors to study</li>
                
            </ul>
        </blockquote>
        

        
        <h1>Useful literature</h1>
        <p>Useful information regarding this type of analysis with descriptions and paper references for the tools used in this tutorial, and literature for this analysis techniques and interpretations can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
        

        <h3>
<i class="fa fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

        <hr>

        <blockquote class="overview">
            <h3>
<i class="fa fa-comments-o" aria-hidden="true"></i> Feedback</h3>
            Please take a moment and provide your feedback on this tutorial. Your feedback will help guide and improve future revisions to this tutorial.
            <a href="https://tinyurl.com/GTNfeedback">Feedback Form</a>
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Bérénice Batut, Mallory Freeberg, Mo Heydarian, Anika Erxleben, Pavankumar Videm, Clemens Blank)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/ref-based/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
</html>
